mlflow/models/python_api.py

```diff
@@ -178,13 +195,41 @@ def _serialize_input_data(input_data, content_type):
             # first row to be the header
             return pd.DataFrame(input_data).to_csv(index=False)
         else:
-            return json.dumps(input_data)
+            return _serialize_to_json(input_data)
     except Exception as e:
         raise MlflowException.invalid_parameter_value(
-            message="Input data could not be serialized to {content_type}."
+            message=f"Input data could not be serialized to {content_type}."
         ) from e
 
 
+def _serialize_to_json(input_data):
+    # imported inside function to avoid circular import
+    from mlflow.pyfunc.scoring_server import SUPPORTED_FORMATS, SUPPORTED_LLM_FORMAT
+
+    if isinstance(input_data, dict) and any(
+        key in input_data for key in SUPPORTED_FORMATS | {SUPPORTED_LLM_FORMAT}
```

Comment:
/ai What does `SUPPORTED_FORMATS | {SUPPORTED_LLM_FORMAT}` mean? `SUPPORTED_FORMATS` is a set.

Reply:
@B-Step62 Can we use `SUPPORTED_FORMATS.union`?

Reply:
hmm but `|` is more concise and readable imo?

Reply:
This is a nit, not a blocker at all. `|` is fine because now I know what it does :)